### PR TITLE
Feature/inject method dependencies

### PR DESCRIPTION
**Pull request by sGeeK44**

### CODE REVIEW

File: nurse/api.py [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import inspect
+from nurse.exceptions import DependencyError
 from .service_catalog import ServiceCatalog
 
 
-def inject(user_class):
+def inject(*decorator_args):
     """
     A decorator that injects dependencies into every instances of a user-defined class.
```
Comment (alexismanuel) [inline nurse/api.py]: Je suggère de mettre à jour ce commentaire: par exemple `A decorator that injects dependencies into every instances of a user-defined class or method`

File: poetry.lock [hunk 2]
```diff
@@ -53,6 +61,36 @@ optional = false
 python-versions = ">=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*"
 version = "0.4.1"
 
+[[package]]
```
Comment (ducdetronquito) [inline poetry.lock]: Are you sure you only added Flake8 ?

File: nurse/exceptions.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class DependencyError(Exception):
+    def __init__(self, message: str):
+        self.message = message
+
+    def __str__(self):
```
Comment (ducdetronquito) [inline nurse/exceptions.py]: Missing return type:
```python
def __str__(self) -> str:
    ...
```

File: nurse/exceptions.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class DependencyError(Exception):
+    def __init__(self, message: str):
```
Comment (ducdetronquito) [inline nurse/exceptions.py]: Missing return type:
```python
def __init__(self) -> None:
    ...
```

File: nurse/api.py [hunk 5]
```diff
@@ -19,23 +19,56 @@ class Server:
...
+            service = service_catalog.services.get(service_type, None)
+            if not service:
+                raise DependencyError(f"Dependency `{service_type}` for `{param_to_inject}` was not found.")
+            return service
 
-    return user_class
+        return decorator
+
+    service_catalog = ServiceCatalog.get_instance()
+    if isinstance(decorator_args[0], str):
```
Comment (ducdetronquito) [inline nurse/api.py]: We should be also able to specify the parameters to inject in a class.

File: poetry.lock (hunk 2 above)
Comment (sGeeK44) [reply poetry.lock]: Yes I'm sure.

File: nurse/api.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 from .service_catalog import ServiceCatalog
 
 
-def inject(*decorator_args):
+def inject(*items_to_inject):
```
Comment (ducdetronquito) [inline nurse/api.py]: Types

```python
def inject(*dependencies: List[str]):
    ...
```

File: README.rst [hunk 7]
```diff
@@ -88,7 +88,7 @@ to any of your class.
 
     import nurse
 
-    @nurse.inject
+    @nurse.inject('player')
```
Comment (ducdetronquito) [inline README.rst]: Missing a dependency + use double quotes :3
```python
@nurse.inject("player", "enemy")
    ...
```

**Pull request merged**